control-plane/connect-inject/controllers/endpoints/endpoints_controller.go

```diff
@@ -999,8 +1008,87 @@ func (r *Controller) deregisterService(apiClient *api.Client, k8sSvcName, k8sSvc
 		}
 	}
 
-	return errs
+	if requeueAfter > 0 {
+		r.Log.Info("re-queueing event", "requeueAfter", requeueAfter)
```

Comment:
should we include service and namespace or other meta?  you could grab it from the log line above, but if you were looking at this in a structured log viewer and quering for requeued events, it might be helpful to have that.  Also, it might be helpful to say `re-queueing <event type> event` to help differentiate it from other requeuese we might do during registration.

Reply:
Good idea. I'll add the meta data and clarify this is for graceful shutdown.